dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs

```diff
@@ -0,0 +1,280 @@
+
+        // Weaviate Memory Store
+        // store = CreateSampleWeaviateMemoryStore();
+
+        // Redis Memory Store
+        // store = await CreateSampleRedisMemoryStoreAsync();
+
+        // Postgres Memory Store
+        // store = await CreateSamplePostgresMemoryStoreAsync();
+
+        // Kusto Memory Store
+        // store = await CreateSampleKustoMemoryStoreAsync();
+
+        await RunWithStoreAsync(store, cancellationToken);
+    }
+
+    private static async Task<IMemoryStore> CreateSampleSqliteMemoryStoreAsync()
+    {
+        IMemoryStore store = await SqliteMemoryStore.ConnectAsync("memories.sqlite");
+        return store;
+    }
+
+    private static async Task<IMemoryStore> CreateSampleDuckDbMemoryStoreAsync()
+    {
+        IMemoryStore store = await DuckDBMemoryStore.ConnectAsync("memories.duckdb");
+        return store;
+    }
+
+    private static IMemoryStore CreateSampleAzureCognitiveSearchMemoryStore()
+    {
+        IMemoryStore store = new AzureCognitiveSearchMemoryStore(TestConfiguration.ACS.Endpoint, TestConfiguration.ACS.ApiKey);
+        return store;
+    }
+
+    private static IMemoryStore CreateSampleQdrantMemoryStore()
+    {
+        IMemoryStore store = new ChromaMemoryStore(TestConfiguration.Chroma.Endpoint, ConsoleLogger.Logger);
+        return store;
+    }
+
+    private static IMemoryStore CreateSampleChromaMemoryStore()
+    {
+        IMemoryStore store = new QdrantMemoryStore(TestConfiguration.Qdrant.Endpoint, 1536, ConsoleLogger.Logger);
+        return store;
+    }
+
+    private static IMemoryStore CreateSamplePineconeMemoryStore()
+    {
+        IMemoryStore store = new PineconeMemoryStore(TestConfiguration.Pinecone.Environment, TestConfiguration.Pinecone.ApiKey, ConsoleLogger.Logger);
+        return store;
+    }
+
+    private static IMemoryStore CreateSampleWeaviateMemoryStore()
+    {
+        IMemoryStore store = new WeaviateMemoryStore(TestConfiguration.Weaviate.Endpoint, TestConfiguration.Weaviate.ApiKey);
+        return store;
+    }
+
+    private static async Task<IMemoryStore> CreateSampleRedisMemoryStoreAsync()
+    {
+        string configuration = TestConfiguration.Redis.Configuration;
+        await using ConnectionMultiplexer connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync(configuration);
+        IDatabase database = connectionMultiplexer.GetDatabase();
+        IMemoryStore store = new RedisMemoryStore(database, vectorSize: 1536);
+        return store;
+    }
+
+    private static async Task<IMemoryStore> CreateSamplePostgresMemoryStoreAsync()
+    {
+        NpgsqlDataSourceBuilder dataSourceBuilder = new(TestConfiguration.Postgres.ConnectionString);
+        dataSourceBuilder.UseVector();
+        await using NpgsqlDataSource dataSource = dataSourceBuilder.Build();
+        IMemoryStore store = new PostgresMemoryStore(dataSource, vectorSize: 1536, schema: "public");
+        return store;
+    }
```

Comment:
Redis and Postgres connectors will throw an exception on a first request, because here we are disposing `ConnectionMultiplexer` in case of Redis and `NpgsqlDataSource` in case of Postgres at the end of these initialization methods. We should initialize them on higher level (maybe at the beginning of `RunAsync` method) and use them here or skip disposal since these are just samples.